src/index.d.ts

```diff
@@ -1 +1,17 @@
-// PRs welcome
+import * as React from 'react';
+
+export default function createReactContext<T>(defaultValue: T): Context<T>;
+
+export type Context<T> = {
+  Provider: React.ComponentClass<ProviderProps<T>>;
+  Consumer: React.ComponentClass<ConsumerProps<T>>;
+};
+
+export type ProviderProps<T> = {
+  value: T;
+  children?: React.ReactNode;
```

Comment:
Why children is optional?

Reply:
Just following the same convention of any component that accepts element children. But it probably should be required, since there's no point in using a provider without any components to consume it. I'll fix it.

Reply:
Also React do not allow to render undefined as children.
https://codepen.io/TrySound/pen/dJqWaW?editors=0010